pkg/kubelet/status/state/state_checkpoint.go

```diff
@@ -43,7 +43,7 @@ func NewStateCheckpoint(stateDir, checkpointName string) (State, error) {
 		return nil, fmt.Errorf("failed to initialize checkpoint manager for pod allocation tracking: %v", err)
 	}
 	stateCheckpoint := &stateCheckpoint{
-		cache:             NewStateMemory(),
+		cache:             NewStateMemory(PodResourceAllocation{}, PodResizeStatus{}),
 		checkpointManager: checkpointManager,
 		checkpointName:    checkpointName,
 	}
```

Comment:
I think the suggestion at #129477 is something like:

```go
        praInfo, err  := restoreState()
        stateCheckpoint.cache = NewStateMemory(praInfo.AllocationEntries)
```

Reply:
nice, like that?